src/tutorials/0004/07.vue

```diff
@@ -15,135 +15,7 @@ import text from './07.md'
-    // user edited the ls line to show something other than root directory, which will also cause most later checks to fail
-    let returnedDirectoryMsg = ''
-    if (listedSome) {
-      returnedDirectoryMsg = ' in your `/some` directory'
-    } else if (listedSomeStuff) {
-      returnedDirectoryMsg = ' in your `/some/stuff` directory'
-    } else {
-      returnedDirectoryMsg = ''
-    }
-    return {
-      fail: 'Looks like you edited the `ls` code to list something other than the root directory. Please try again, editing only the section of code indicated.',
-      logDesc: 'Here\'s what your `ls` command shows' + returnedDirectoryMsg + ':',
-      log: logResult
-    }
-  } else if (uploadedFiles === false) {
-    // shouldn't happen because you can't hit submit without uploading files
-    return {fail: 'Oops! You forgot to upload files to work with :('}
-  } else if (!rootSomeItemIsFile) {
-    // no files written to IPFS because user changed default code to remove write command
-    return {
-      fail: 'Uh oh. Looks like no files made it into IPFS. Did you accidentally edit the default `write` code?',
-      logDesc: "Here's what's in your root directory:",
-      log: logRoot
-    }
-  } else if (contains(stuffSomeHash, 'stuff')) {
-    // created /stuff/some instead of /some/stuff
-    return { fail: 'Uh oh. Looks like you created /stuff/some instead of `/some/stuff`.' }
-  } else if (contains(emptyDirectoryHash, 'stuff')) {
-    // created /stuff instead of /some/stuff.
-    return { fail: 'Uh oh. Looks like you created /stuff instead of `/some/stuff`.' }
-  } else if (contains(emptyDirectoryHash, 'some')) {
-    // created /some instead of /some/stuff
-    return { fail: 'Uh oh. Looks like you created /some instead of `/some/stuff`.' }
-  } else if (!contains(someStuffHash, 'some')) {
-    // didn't create empty some/stuff
-    return { fail: 'Uh oh. Looks like your directory doesn\'t contain an empty `/some/stuff/` directory.' }
-  } else if (contains(someStuffHash, 'some') && !contentsMatch) {
-    // created empty /some/stuff but other files are wrong (messed up write method)
-    return {
-      fail: 'Hmmm. You created a `/some/stuff` directory but something else is wrong. Did you accidentally edit the default `write` code so your other files weren\'t all added?',
-      logDesc: "Here's what was returned by `ls` in your root directory.",
-      log: logRoot
-    }
-  } else if (contains(someStuffHash, 'some') && contentsMatch) {
-    // filenames match and created empty some/stuff
-    return {
-      success: 'Success! Check out your directory contents below.',
-      logDesc: "Here's what was returned by `ls` in your root directory. Notice how directories have a type of `1` while files have a type of `0`.",
-      log: logRoot
-    }
-  } else if (result.error) {
-    return { error: result.error }
-  } else {
-    return { fail: 'Something we haven\'t anticipated is wrong. :(' }
-  }
+  return null
```

Comment:
It looks like someone accidentally committed the version of `src/tutorials/0004/07.vue` that we changed for testing, replacing all of the validation code with `return null`. We could copy and paste the pre-this-PR version to replace this, but I'm not sure if there were other intentional changes made to this file before the accidental one. @zebateira @dominguesgm does one of you know the best way to address this on GitHub or locally? Is there a way to revert just one file's worth of changes one commit at a time until it goes back to having the last reasonable content? I'd love to be taught a good approach to this if either of you have any insight.

Reply:
Oops, my bad, great catch @terichadbourne 😅.

The best approach in this case (other than simply checking the past commit version and copy-paste it), is to checkout the previous version of the file with this command:

`git checkout HEAD~1 -- src/tutorials/0004/07.vue`

Btw, I also just remembered I had somewhat tackled issue #337 here. @zebateira do you want to give some feedback if you think the change made in mfs' lesson 7, line 81, is enough to fix the issue?